test/AllPairs/test_ap.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+import docker
```

Comment:
```suggestion
import shutil
```
It doesn't look like we use docker directly, and shutil will resolve the testing error.